projects/demo-integrations/src/tests/component-testing/phone/phone-with-initial-value.cy.ts

```diff
@@ -0,0 +1,395 @@
+                    .should('have.prop', 'selectionEnd', '+7 771 931'.length)
+                    .type('{backspace}')
+                    .should('have.value', '+7 771 931-111')
+                    .should('have.prop', 'selectionStart', '+7 771 93'.length)
+                    .should('have.prop', 'selectionEnd', '+7 771 93'.length);
+            });
+
+            it('+7 77|1 931-1111 => Backspace => +7 7|19 311-111', () => {
+                cy.get('input')
+                    .should('be.visible')
+                    .should('have.value', '+7 771 931-1111')
+                    .focus()
+                    .type('{moveToEnd}')
+                    .type('{leftArrow}'.repeat('1 931-1111'.length))
+                    .should('have.prop', 'selectionStart', '+7 77'.length)
+                    .should('have.prop', 'selectionEnd', '+7 77'.length)
+                    .type('{backspace}')
+                    .should('have.value', '+7 719 311-111')
+                    .should('have.prop', 'selectionStart', '+7 7'.length)
+                    .should('have.prop', 'selectionEnd', '+7 7'.length);
+            });
+        });
+
+        describe('Backspace after typing (confirms mask works when value is typed)', () => {
+            beforeEach(() => {
+                cy.mount(TestInput, {
+                    componentProperties: {
+                        maskitoOptions: createMaskitoOptions(),
+                        initialValue: '+7 ',
+                    },
+                });
+            });
+
+            it('Type value, then backspace in middle works correctly', () => {
+                cy.get('input')
+                    .should('be.visible')
+                    .focus()
+                    .type('7719311111')
+                    .should('have.value', '+7 771 931-1111')
+                    // Now test backspace in the middle
+                    .type('{leftArrow}'.repeat('31-1111'.length))
+                    .should('have.prop', 'selectionStart', '+7 771 9'.length)
+                    .should('have.prop', 'selectionEnd', '+7 771 9'.length)
+                    .type('{backspace}')
+                    .should('have.value', '+7 771 311-111')
+                    .should('have.prop', 'selectionStart', '+7 771 '.length)
+                    .should('have.prop', 'selectionEnd', '+7 771 '.length);
+            });
+        });
+    });
+
+    describe('Strict mode (United States)', () => {
+        function createMaskitoOptions(): MaskitoOptions {
+            return maskitoPhoneOptionsGenerator({
+                countryIsoCode: 'US',
+                metadata,
+                strict: true,
+            });
+        }
+
+        describe('Backspace on initial render', () => {
+            beforeEach(() => {
+                cy.mount(TestInput, {
+                    componentProperties: {
+                        maskitoOptions: createMaskitoOptions(),
```

Comment:
Is this function is used only once ?

Could we just inline it ?

```suggestion
        describe('Backspace on initial render', () => {
            beforeEach(() => {
                cy.mount(TestInput, {
                    componentProperties: {
                        maskitoOptions: maskitoPhoneOptionsGenerator({
                            countryIsoCode: 'US',
                            metadata,
                            strict: true,
                        }),
```

Reply:
Great catch, done 💪